default.nix

```diff
@@ -1,22 +1,73 @@
-{ pkgs ? import <nixpkgs> { }
-, hypothesisProfile ? null
-, collectCoverage ? false
-, testSuite ? null
-, trialArgs ? null
-, tahoe-lafs ? "1.14.0"
-}:
 let
-  tahoe-packages = {
-    "1.14.0"    = pkgs.python2Packages.tahoe-lafs-1_14;
-    "1.16.0rc1" = pkgs.python2Packages.callPackage ./nix/tahoe-lafs-1_16.nix { };
-  };
-  tahoe-lafs' = builtins.getAttr tahoe-lafs tahoe-packages;
-
-  pkgs' = pkgs.extend (import ./overlays.nix);
-  callPackage = pkgs'.python27Packages.callPackage;
+  sources = import nix/sources.nix;
 in
-callPackage ./zkapauthorizer.nix {
-  challenge-bypass-ristretto = callPackage ./python-challenge-bypass-ristretto.nix { };
-  inherit hypothesisProfile collectCoverage testSuite trialArgs;
-  tahoe-lafs = tahoe-lafs';
-}
+{ pkgs ? import sources.release2015 {}
+, pypiData ? sources.pypi-deps-db
+, mach-nix ? import sources.mach-nix { inherit pkgs pypiData; }
+, tahoe-lafs-repo ? sources.tahoe-lafs
+}:
+  let
+    lib = pkgs.lib;
+    providers = {
+      _default = "sdist,nixpkgs,wheel";
+      # mach-nix doesn't provide a good way to depend on mach-nix packages,
+      # so we get it as a nixpkgs dependency from an overlay. See below for
+      # details.
+      tahoe-lafs = "nixpkgs";
+      # not packaged in nixpkgs at all, we can use the binary wheel from
+      # pypi though.
+      python-challenge-bypass-ristretto = "wheel";
+      # Pure python packages that don't build correctly from sdists
+      # - patches in nixpkgs that don't apply
+      boltons = "wheel";
+      chardet = "wheel";
+      urllib3 = "wheel";
+      # - incorrectly detected dependencies due to pbr
+      fixtures = "wheel";
+      testtools = "wheel";
+      traceback2 = "wheel";
+    };
+    tahoe-lafs = mach-nix.buildPythonPackage rec {
+      python = "python27";
+      name = "tahoe-lafs";
+      version = "1.16.post999";
```

Comment:
This will need to be manually bumped at some point.

It'd be nice if tahoe supported version from github tarballs. git now has a `%(describe)` format, which could be used something like [this](https://github.com/tahoe-lafs/tahoe-lafs/compare/master...tp-la:git-archive-version), but it appears that github doesn't currently support that option.

Reply:
Tahoe could support simpler versions from github tarballs, though, right?